harness/determined/common/experimental/__init__.py

```diff
@@ -5,3 +5,13 @@
 from determined.common.experimental.experiment import ExperimentReference
 from determined.common.experimental.trial import TrialReference, TrialSortBy, TrialOrderBy
 from determined.common.experimental.model import Model, ModelOrderBy, ModelSortBy, ModelVersion
+
+import warnings
+
+warnings.warn(
+    "The 'determined.common.experimental' module is deprecated and will be removed "
```

Comment:
`determined.common` is 100% internal and no end-users should ever use it, ever.

There is a boring legacy reason why this code was originally where it was.

The reason I didn't deprecate it the symbols in d.c.e when I added the `experimental.client` module was because I expected that we'd be promoting the python sdk out of experiment like... years ago... so I figured I'd do all the breaking changes in one go, no need for waves of deprecation.

In reality, the python sdk has always been half-baked until we put real engineering resources on it this quarter, so it wasn't ever promoted.

My conclusion is: external users running really old code might be using d.c.e and it's worth having a warning.

Internal users can write in d.c.e today and move it tomorrow without any warning and that's fine.  External users of today should be accessing all these symbols from `det.experimental.client` and in the future from `det.client` regardless of where new symbols are added.